src/solver/algorithm/strong/cyclic.rs

```diff
@@ -6,3 +6,379 @@
 //!
 //! #### Authorship
 //! - Max Fierro, 12/3/2023 (maxfierro@berkeley.edu)
+//! - Ishir Garg, 3/12/2024 (ishirgarg@berkeley.edu)
+
+use anyhow::{Context, Result};
+use std::collections::{HashMap, VecDeque};
+
+use crate::database::volatile;
+use crate::database::{KVStore, Tabular};
+use crate::game::{Bounded, DTransition, Extensive, Game, SimpleSum};
+use crate::interface::IOMode;
+use crate::model::SimpleUtility;
+use crate::model::{PlayerCount, Remoteness, State, Turn};
+use crate::solver::record::sur::RecordBuffer;
+use crate::solver::RecordType;
+
+/* CONSTANTS */
+
+/// The exact number of bits that are used to encode remoteness.
+const REMOTENESS_SIZE: usize = 16;
+
+/// The maximum number of bits that can be used to encode a record.
+const BUFFER_SIZE: usize = 128;
+
+/// The exact number of bits that are used to encode utility for one player.
+const UTILITY_SIZE: usize = 2;
+
+pub fn two_player_zero_sum_dynamic_solver<G>(
+    game: &G,
+    mode: IOMode,
+) -> Result<()>
+where
+    G: DTransition<State> + Bounded<State> + SimpleSum<2> + Extensive<2> + Game,
+{
+    let mut db =
+        volatile_database(game).context("Failed to initialize database.")?;
+    basic_loopy_solver(game, &mut db)?;
+    Ok(())
+}
+
+fn basic_loopy_solver<G, D>(game: &G, db: &mut D) -> Result<()>
+where
+    G: DTransition<State> + Bounded<State> + SimpleSum<2> + Extensive<2> + Game,
+    D: KVStore,
+{
+    let mut winning_frontier = VecDeque::new();
+    let mut tying_frontier = VecDeque::new();
+    let mut losing_frontier = VecDeque::new();
+
+    let mut child_counts = HashMap::new();
```

Comment:
Eventually we'll want to use an equivalent of the `surcc` record implementation by instantiating a new database table, not a HashMap (for reasons discussed on Slack).

This is good for now, since we haven't worked out the semantics of having many database tables active at the same time.